src/txn-wal/src/txns.rs

```diff
@@ -109,11 +110,13 @@ where
     O: Opaque + Debug + Codec64,
     C: TxnsCodec,
 {
+    pub(crate) dyncfgs: ConfigSet,
     pub(crate) metrics: Arc<Metrics>,
     pub(crate) txns_cache: TxnsCache<T, C>,
     pub(crate) txns_write: WriteHandle<C::Key, C::Val, T, i64>,
     pub(crate) txns_since: SinceHandle<C::Key, C::Val, T, i64, O>,
-    pub(crate) datas: DataHandles<K, V, T, D>,
+    pub(crate) data_write_apply: DataHandles<K, V, T, D>,
+    pub(crate) data_write_commit: BTreeMap<ShardId, DataWriteCommit<K, V, T, D>>,
```

Comment:
Do you mind adding a comment explaining the difference between these two? Mainly when would a handle existing in `data_write_apply` and not `data_write_commit`, or vice versa? Could a shard ever have a write handle in both?

Reply:
Done!